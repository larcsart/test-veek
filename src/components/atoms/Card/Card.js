import React from 'react';
import PropTypes from 'prop-types';
import styles from './Card.module.scss';

const Card = ({ children }) => (
  <div className={styles.Card} data-testid="Card">
    {children}
  </div>
);

Card.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Card;
