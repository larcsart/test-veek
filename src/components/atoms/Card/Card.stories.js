/* eslint-disable */
import React from 'react';
import { storiesOf } from '@storybook/react';
import Card from './Card';
import { ProductCard } from '~/components/molecules';

storiesOf('Card', module).add('default', () => (
  <Card>
    <ProductCard
      imgUrl="https://http2.mlstatic.com/D_NQ_NP_600943-MLB31871248861_082019-V.webp" 
      price={{
        currency: 'R$',
        amount: 1000,
        decimals: 12,
      }}
      cityName="Rio de Janeiro"
      freeShipping
      name="Bateria Notebook - Dell Inspiron 14 3442 (14.8v) - Preta"
    />
  </Card>
));
