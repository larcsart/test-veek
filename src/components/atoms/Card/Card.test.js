import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Card from './Card';

describe('<Card />', () => {
  test('it should mount', () => {
    render(
      <Card><div /></Card>,
    );

    const card = screen.getByTestId('Card');

    expect(card).toBeInTheDocument();
  });

  test('snapshot', () => {
    const { asFragment } = render(
      <Card><div /></Card>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
