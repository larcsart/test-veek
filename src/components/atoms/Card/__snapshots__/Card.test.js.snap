// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
    data-testid="Card"
  >
    <div />
  </div>
</DocumentFragment>
`;
